docs/reference/cmodules.rst

```diff
@@ -0,0 +1,86 @@
+Extending MicroPython with C
+============================
+
+Some specialized code would be unacceptably slow or needs to access hardware in
+a way that cannot be done from MicroPython. Therefore, it supports a way of
+extending the language with custom modules written in C. But before you consider
+writing a module in C, please take a look at :ref:`speed_python`.
+
+`Unlike CPython <https://docs.python.org/3/extending/building.html>`_, these
+modules are (currently) embedded directly in the program image instead of being
+dynamically loaded. This requires a `custom build of MicroPython
+<https://github.com/micropython/micropython/wiki/Getting-Started>`_.
+
+
+Writing a module
+----------------
+
+A module is a directory with the following files:
+
+  * ``micropython.mk``, which contains the Makefile fragment for this module.
+  * All C files you would like included.
+
+Put the required build commands in ``micropython.mk``. For a simple module, you
+will only have to add the file paths to ``SRC_MOD``, which will include these C
+files in the build:
+
+.. highlight:: make
+.. code::
+
+    # Add all C files to SRC_MOD.
+    SRC_MOD += $(MODULE_DIR)/example/example.c
```

Comment:
It worked a bit better with USER_C_MODULES instead of MODULE_DIR.

Reply:
Oops, my bad. I need to fix the docs there.

Reply:
Fixed!